src/containerapp/azext_containerapp/_cloud_build_utils.py

```diff
@@ -129,28 +129,34 @@ def spin():
         thread.join()
 
         # Source code compression
-        done_spinner = False
-        thread = display_spinner(f"Compressing data: {font_bold}{source}{font_default}")
-        tar_file_path = os.path.join(tempfile.gettempdir(), f"{build_name}.tar.gz")
-        archive_source_code(tar_file_path, source)
-        done_spinner = True
-        thread.join()
+        data_file_path = source
+        source_is_folder = os.path.isdir(source)
+        if source_is_folder:
+            done_spinner = False
+            thread = display_spinner(f"Compressing data: {font_bold}{source}{font_default}")
+            data_file_path = os.path.join(tempfile.gettempdir(), f"{build_name}.tar.gz")
+            archive_source_code(data_file_path, source)
+            done_spinner = True
+            thread.join()
 
         # File upload
         done_spinner = False
-        thread = display_spinner("Uploading compressed data")
+        thread = display_spinner("Uploading data")
         headers = {'Authorization': 'Bearer ' + token}
         try:
-            tar_file = open(tar_file_path, "rb")
-            files = [("file", ("build_data.tar.gz", tar_file, "application/x-tar"))]
+            data_file = open(data_file_path, "rb")
+            file_name = os.path.basename(data_file_path)
+            files = [("file", (file_name, data_file))]
```

Comment:
Just wanted to clarify this change: why couldn't we keep the previous logic to generate a new file name and push that?
The part that makes me double think is that if we accept any file name here, I'm not sure how well we support names with accents, etc. on the Cloud Build side.

Reply:
Hi @daniv-msft , the reason why we changed the logic here is because for special file extensions like tar.gz, we cannot easily split it from file name. RP code will use the suffix of the file name to validate file extension. Keeping customer's original data_file name can also help maintain original data.